launcher/resources_library/utils/operations.go

```diff
@@ -0,0 +1,382 @@
+	containers, err := cli.ContainerList(ctx, container.ListOptions{All: true})
+	if err != nil {
+		return err, false
+	}
+
+	containerMap := make(map[string]string)
+	for _, container := range containers {
+		for _, name := range container.Names {
+			containerMap[name] = strings.ToLower(container.State)
+		}
+	}
+
+	state, exists := containerMap["/"+containerName]
+	if !exists {
+		return nil, false
+	}
+
+	return nil, state == "running"
+}
+
+func removeContainer(ctx context.Context, cli *client.Client, containerID string) error {
+	return cli.ContainerRemove(ctx, containerID, container.RemoveOptions{Force: true})
+}
+
+func CreateAndRunContainer(ctx context.Context, cli *client.Client, log logr.Logger, containerInfo general.Containers) error {
+	err, isRunning := isContainerRunning(ctx, cli, containerInfo.ContainerName)
+	if err != nil {
+		log.Error(err, "Failed to read about status of container")
+		return err
+	}
+
+	if isRunning {
+		log.Info("Container ", containerInfo.ContainerName, " is runnning. Omittig this container creation.")
+		return nil
+	}
+
+	err, exists := doesContainerExist(ctx, cli, containerInfo.ContainerName)
+	if err != nil {
+		log.Error(err, "Failed to read about status of container")
+		return err
+	}
+
+	if exists {
+		log.Info("Removing container to re-create and re-run because container with a such name exists but with status exited:", "container", containerInfo.ContainerName)
+		err = removeContainer(ctx, cli, containerInfo.ContainerName)
+		if err != nil {
+			log.Error(err, "Failed to remove container")
+			return err
+		}
+
+	}
+
+	err = pullImageIfNotExists(ctx, cli, containerInfo.Image, log)
+	if err != nil {
+		log.Error(err, "Error pulling image for container")
+		return err
+	}
+	// Define the container configuration
+	containerConfig := &container.Config{
+		// Tty:   true,
+		Image: containerInfo.Image,
+		ExposedPorts: nat.PortSet{
+			nat.Port(containerInfo.ExposedPort): struct{}{},
+		},
+	}
+
+	// Define the host configuration
+	hostConfig := &container.HostConfig{
+		NetworkMode: container.NetworkMode(general.NetworkModeHost),
+		PortBindings: nat.PortMap{
+			nat.Port(containerInfo.ExposedPort): []nat.PortBinding{
+				{
+					HostIP:   containerInfo.Ip,
+					HostPort: containerInfo.BindingHostPort,
+				},
+			},
+		},
+	}
+
+	// Define the network configuration
+	networkConfig := &network.NetworkingConfig{}
+
+	// Create the container
+	resp, err := cli.ContainerCreate(ctx, containerConfig, hostConfig, networkConfig, nil, containerInfo.ContainerName)
+
+	if err != nil {
+		log.Error(err, "Error creating container")
+		return err
+	}
+
+	// Start the container
+	err = cli.ContainerStart(ctx, resp.ID, container.StartOptions{})
+	if err != nil {
+		log.Error(err, "Error starting container")
+		return err
+	}
+
+	statusCh, errCh := cli.ContainerWait(ctx, resp.ID, container.WaitConditionNotRunning)
+	select {
+	case err := <-errCh:
```

Comment:
You don't care about the input context cancellation, do you?
```suggestion
	select {
	case <-ctx.Done():
		// handle appropriately
	case err := <-errCh:
```